### PR TITLE
WIP Basic group moving

### LINKED ISSUE #956: Enable resizing of Groups
__Is your feature request related to a problem? Please describe.__

Right now, it is not possible to resize a group in the diagram.

__Describe the solution you'd like__

Make it possible to resize groups. 

__Additional context__

* Hint: Modify [`canResize`](https://github.com/bpmn-io/bpmn-js/blob/master/lib/features/rules/BpmnRules.js#L751) rule.

---

Child of #345

### LINKED ISSUE #959: It is not possible to move elements into a group
__Describe the Bug__

It is currently not possible to move elements into a group.

Mar-27-2019 10-49-30

__Steps to Reproduce__

1. Move an element into a group element
2. Dropping is not allowed

__Expected Behavior__

It should be possible to move all kind of elements into groups.

__Environment__

Please complete the following information:

 - Browser: Chrome 72
- OS: macOS 10.14
- Library version: v3.2.3

---

Child of #343

### PR DESCRIPTION
**Pull request by pinussilvestrus**
This pull request starts the general implementation of `bpmn:Group` support. Therefore it fixes the following issues:

- Closes #959 
- Closes #956 

Apr-03-2019 08-49-28

**Note:** The changes come with several modeling trade-offs regarding group elements for now, e.g.

Apr-03-2019 08-55-21

One approach would be to simply increase the order of a group element (e.g. in this [example branch](https://github.com/bpmn-io/bpmn-js/commits/group-experiments)). But, since group elements have always to act in the background and they have to be rendered on top otherwise, will we need to implement a completely new type of shapes, which are only draggable in the background and are kind of empty inside (cf. #960, #957). This will follow in another pull request, which needs the current changes as its base.

### CODE REVIEW

File: lib/features/rules/BpmnRules.js [hunk 1]
```diff
@@ -786,6 +786,10 @@ function canResize(shape, newBounds) {
     return true;
   }
 
+  if (isGroup(shape)) {
```
Comment (nikku) [inline lib/features/rules/BpmnRules.js]: Could we also add a test case that verifies that groups are properly resized in the BPMN 2.0 XML?

**Comment by pinussilvestrus** [comment]
Although we decided to go the _frame_ way (btw. I really like this description!), I gave this changes to review for 2 reasons:

* we will have to add resize support for groups otherwise in future
* right now, I am not completely sure whether we will have to deal with the group dropping after we implemented the necessary stuff in diagram-js. Of course the aim is to act groups as a frame (just interactable on borders, no other modeling interactions at all). But I was simply not sure what then happens when we, for example, drop elements on the border of a group. It depends on the final implementation in diagram-js. My personal goal would be to get rid of the current behavior introduced in this pull request.

Proposal: Put the WIP label on this pull request and just wait for the diagram-js implementation. I use this branch as a testing branch since I need some interaction support to verify the implementation. We could also close this for the moment an eventually re-open it.

**Comment by nikku** [comment]
Thanks for the clarification :+1:.

I've added a WIP label to avoid confusion.

**Comment by pinussilvestrus** [comment]
Update: We don't need these changes since we're going to _not allow_ dropping elements on group elements (cf. https://github.com/bpmn-io/diagram-js/pull/321). I'll close this and will open another one with integrating the new frame element support. Next to the resize support (cf. #956), this will contain additional stuff we can do with frame elements. Stay tuned 📻

**Pull request closed without merging**